tests/end_to_end/conftest.py

```diff
@@ -195,3 +196,48 @@ def relayers(worker_id):
     except ValueError:
         logger.info(f"Error while setting relayer index to {worker_id}")
     return
+
+
+# Uniswap fixtures
+
+TOTAL_SUPPLY = 10000 * 10**18
+
+
+@pytest_asyncio.fixture(scope="function")
+async def token_a(owner):
+    return await deploy_kakarot(
+        "UniswapV2",
+        "ERC20",
+        TOTAL_SUPPLY,
+        caller_eoa=owner.starknet_contract,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def weth(owner):
+    return await deploy_kakarot(
+        "WETH",
+        "WETH9",
+        caller_eoa=owner.starknet_contract,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def factory(owner):
+    return await deploy_kakarot(
+        "UniswapV2",
+        "UniswapV2Factory",
+        owner.address,
+        caller_eoa=owner.starknet_contract,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def router(owner, factory, weth):
+    return await deploy_kakarot(
+        "UniswapV2Router",
+        "UniswapV2Router02",
+        factory.address,
+        weth.address,
+        caller_eoa=owner.starknet_contract,
```

Comment:
for all these tokens, it `msg.sender` is not used in the constructor, remove the `caller_eao` kwarg to make it clear that it's doesn't matter